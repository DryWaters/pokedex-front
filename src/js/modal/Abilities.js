import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/modal/Abilities.css';

const propTypes = {
  abilities: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      hidden: PropTypes.bool.isRequired,
    }).isRequired,
  ).isRequired,
};

class Abilities extends React.Component {
  render () {
    const abilityElements = this.props.abilities.map((ability, index) => { return  ability.hidden ? null : (
      <span key={`container-${ability.name}`}>
        {index > 0 ? ', ' : ''}
        <span className="ability">
          {ability.name}
          <span className="ability-tooltip">{ability.description}</span>
        </span>
      </span>
    )});

    return (
      <h4>
        {'Abilities: '}
        {abilityElements}
      </h4>
    );
  }
}

Abilities.propTypes = propTypes;

export default Abilities;
