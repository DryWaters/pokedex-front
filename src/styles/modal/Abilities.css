.ability-tooltip {
  visibility: hidden;
  padding: 20px;
  border-radius: 10px;
  background-color: grey;
  position: absolute;
  bottom: 50px;
  z-index: 1;
  display: table-cell;
  vertical-align: center;
}

.ability:hover .ability-tooltip {
  visibility: visible;
}
