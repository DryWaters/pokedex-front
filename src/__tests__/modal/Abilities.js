import React from 'react';
import { shallow } from 'enzyme';

import Abilities from '../../js/modal/Abilities';

import ExampleJSON from '../../../local_data/example_detail.json';

describe('Abilities Component', () => {
  let props;
  beforeEach(() => {
    props = {
      abilities: ExampleJSON.forms[0].abilities,
    };
  });

  it('should render Abilities component without crashing', () => {
    shallow(<Abilities {...props} />);
  });

  it('should match Abilities component snapshot', () => {
    const wrapper = shallow(<Abilities {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
