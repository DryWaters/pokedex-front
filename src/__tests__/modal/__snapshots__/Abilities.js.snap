// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abilities Component should match Abilities component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Abilities
    abilities={
        Array [
            Object {
              "description": "Increases moves' accuracy to 1.3×.",
              "hidden": false,
              "name": "Compound Eyes",
            },
            Object {
              "description": "Doubles damage inflicted with not very effective moves.",
              "hidden": true,
              "name": "Tinted Lens",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Abilities: ",
        Array [
          <span>
            
            <span
                        className="ability"
            >
                        Compound Eyes
                        <span
                                    className="ability-tooltip"
                        >
                                    Increases moves' accuracy to 1.3×.
                        </span>
            </span>
</span>,
          null,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      "Abilities: ",
      Object {
        "instance": null,
        "key": "container-Compound Eyes",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="ability"
>
              Compound Eyes
              <span
                            className="ability-tooltip"
              >
                            Increases moves' accuracy to 1.3×.
              </span>
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Compound Eyes",
                <span
                  className="ability-tooltip"
>
                  Increases moves' accuracy to 1.3×.
</span>,
              ],
              "className": "ability",
            },
            "ref": null,
            "rendered": Array [
              "Compound Eyes",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Increases moves' accuracy to 1.3×.",
                  "className": "ability-tooltip",
                },
                "ref": null,
                "rendered": "Increases moves' accuracy to 1.3×.",
                "type": "span",
              },
            ],
            "type": "span",
          },
        ],
        "type": "span",
      },
      null,
    ],
    "type": "h4",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Abilities: ",
          Array [
            <span>
              
              <span
                            className="ability"
              >
                            Compound Eyes
                            <span
                                          className="ability-tooltip"
                            >
                                          Increases moves' accuracy to 1.3×.
                            </span>
              </span>
</span>,
            null,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        "Abilities: ",
        Object {
          "instance": null,
          "key": "container-Compound Eyes",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="ability"
>
                Compound Eyes
                <span
                                className="ability-tooltip"
                >
                                Increases moves' accuracy to 1.3×.
                </span>
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Compound Eyes",
                  <span
                    className="ability-tooltip"
>
                    Increases moves' accuracy to 1.3×.
</span>,
                ],
                "className": "ability",
              },
              "ref": null,
              "rendered": Array [
                "Compound Eyes",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Increases moves' accuracy to 1.3×.",
                    "className": "ability-tooltip",
                  },
                  "ref": null,
                  "rendered": "Increases moves' accuracy to 1.3×.",
                  "type": "span",
                },
              ],
              "type": "span",
            },
          ],
          "type": "span",
        },
        null,
      ],
      "type": "h4",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
